frontend/packages/shared/src/components/TreeView/TreeViewItem/TreeViewItem.test.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import type { TreeViewItemProps } from './TreeViewItem';
+import { act, ByRoleOptions, render as renderRtl, screen } from '@testing-library/react';
+import { TreeViewItem } from './TreeViewItem';
+import { TreeViewRootContext, TreeViewRootContextProps } from '../TreeViewRoot';
+import userEvent from '@testing-library/user-event';
+
+const user = userEvent.setup();
+
+// Test data:
+const label = 'label';
+const nodeId = 'nodeId';
+const defaultProps: TreeViewItemProps = {
+  label,
+  nodeId,
+};
+const rootId = 'rootId';
+const setFocusedId = jest.fn();
+const setSelectedId = jest.fn();
+const focusableId = 'focusableId';
+const defaultRootContextProps: TreeViewRootContextProps = {
+  rootId,
+  setFocusedId,
+  setSelectedId,
+  focusableId,
+};
+
+const render = (
+  props: Partial<TreeViewItemProps> = {},
+  rootContextProps: Partial<TreeViewRootContextProps> = {}
+) =>
+  renderRtl(
+    <TreeViewRootContext.Provider value={{ ...defaultRootContextProps, ...rootContextProps }}>
+      <TreeViewItem {...defaultProps} {...props} />
+    </TreeViewRootContext.Provider>
+  );
+
+describe('TreeViewItem', () => {
+  afterEach(jest.clearAllMocks);
+
+  it('Renders a treeitem component with the given label', () => {
+    render();
+    expect(getTreeitem()).toBeInTheDocument();
```

Comment:
Jeg antar at denne koden bare er litt for next level for meg, men jeg skjønner ikke helt hvilken `given label` det er snakk om i denne testen?

Reply:
Det er label fra defaultProps. getTreeitem er satt opp til å finne den hvis ikke noe annet er gitt. Jeg skal se om jeg kan gjøre dette mer tydelig 🙂

Reply:
Fikset